capstone/src/main/java/com/google/univiz/scorecard/SuggestionDataApiImpl.java

```diff
@@ -1,12 +1,34 @@
 package com.google.univiz.scorecard;
 
-import com.google.univiz.api.SuggestionDataApi;
+import com.google.gson.Gson;
 import com.google.univiz.api.representation.SuggestionResponse;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import javax.inject.Inject;
 
 final class SuggestionDataApiImpl implements SuggestionDataApi {
+  private final URLProvider urlProvider;
+  private final CollegeIdReaderProvider readerProvider;
+  private final Gson gson;
+
+  @Inject
+  SuggestionDataApiImpl(
+      URLProvider urlProvider, CollegeIdReaderProvider readerProvider, Gson gson) {
+    this.urlProvider = urlProvider;
+    this.readerProvider = readerProvider;
+    this.gson = gson;
+  }
+
+  /** Takes REST API Json response and converts it to SuggestionData */
+  private SuggestionResponse convertJsonToSuggestionResponse(InputStreamReader suggestionReader) {
+    return gson.fromJson(suggestionReader, SuggestionResponse.class);
+  }
```

Comment:
Per [the style guide](https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents), class members should follow a logical and consistent ordering. Inside google3 we have more strict and clear guideline but the public style guide is not super clear. So this is just a soft suggestion.

Here are few rules you can use for method ordering:
 + Keep constructors at the top. Starting with the default one. Private constructors are exempt.
 + Followed by public methods, then protected, and finally private.
 + Within each visibility group keep static and non-statics in their own sub-group as much as possible.
 + A caller method should usually be above the callee.
 + Keep overloads always together.
 + Keep methods that are working together to accomplish a task closer to each other.

These rules really help the next person who is reading the class from top to bottom.